bin/convert2vulyk.py

```diff
@@ -42,116 +83,173 @@ def convert_bsf_2_vulyk(text: str, bsf_markup: str) -> dict:
-              "normalizations": [], "attributes": [], "events": [], "document": "", "collection": "/"}
+    vulyk = {
+        "modifications": [],
+        "equivs": [],
+        "protocol": 1,
+        "ctime": ts,
+        "triggers": [],
+        "text": text,
+        "source_files": ["ann", "txt"],
+        "messages": [],
+        "sentence_offsets": s_offsets,
+        "comments": [],
+        "entities": ents,
+        "mtime": ts,
+        "relations": [],
+        "token_offsets": t_offsets,
+        "action": "getDocument",
+        "normalizations": [],
+        "attributes": [],
+        "events": [],
+        "document": "",
+        "collection": "/",
+    }
 
     return vulyk
 
 
-def text_2_vulyk(text: str, bsf_markup: str = None) -> str:
-    """
-    Convert text string with annotations to json string for Vulyk annotation tool.
-    If annotations are provided in bsf_markup then text must be tokenized. Otherwise a raw text string can be supplied.
-    :param text: tokenzied OR raw text string
-    :param bsf_markup: string containing annotations in brat standoff format
-    :return: json string compatible with Vulyk (Brat annotation tool)
-    """
-    txt = text
-    markup = bsf_markup
-    if bsf_markup is None:
-        log.info("No markup file => processing text with Stanza to extract Named Entities.")
-        # run tokenization and NER
-        from tokenize_uk import tokenize_text
-        token_list = tokenize_text(text)
-        # we have list<paragraphs> of list<sentences> of list<tokens>
-        paragraph = ['\n'.join([' '.join(t) for t in sent]) for sent in token_list]
-        txt = '\n'.join(paragraph)  # stanza bug does not allow for double new line symbol right now
-        log.debug(txt)
+def convert(args):
+    for text in map(pathlib.Path, glob.glob(args.text_files)):
+        log.info(f"Found text file {text}, parsing it")
 
-        markup = _run_ner(txt)
-        log.debug(markup)
+        markup = ""
 
-    vulyk_obj = convert_bsf_2_vulyk(txt, markup)
+        if not args.ignore_annotations:
+            if args.ann_autodiscovery == "append":
+                ann = text.with_name(text.name + ".ann")
+            else:
+                ann = text.with_suffix(".ann")
 
-    return json.dumps(vulyk_obj, ensure_ascii=False)
+            if not ann.exists():
+                log.warning(f"Cannot find annotation file {ann} alongside to text file {text}, skipping")
+            else:
+                markup = ann.read_text()
 
+        vulyk_obj = convert_bsf_2_vulyk(text.read_text(), markup)
+        print(json.dumps(vulyk_obj, ensure_ascii=False, sort_keys=True))
 
-def _run_ner(txt: str) -> str:
-    """
-    Run ner pipeline on tokenized text using Stanza.
-    :param txt: tokenized text string
-    :return: string with annotations in brat standoff format
-    """
-    import stanza
-    logging.getLogger("stanza").setLevel(log.level)
 
-    stanza.download('uk')
+def tag(args):
```

Comment:
It would be great to have this function with non-generic args.
And a couple of lines of explanation of what function and parameters do - would be fantastic.